### PR TITLE
feat : Added Daily Challenges and implement map-images in commit history

### LINKED ISSUE #15: Daily Challenges

### LINKED ISSUE #19: Map History Images

### PR DESCRIPTION
**Pull request by shubham-1806**
## In this PR
- Updated code and map apis
- Map images will be stored instead map arrays in commits
- DCs code and Normal Code will be saved separately now
- Daily challenge UI
- Daily challenge in Battle TV
- Fix the overflow issue

### CODE REVIEW

**Review by vigneshd332 — APPROVED** [review]

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 1]
```diff
@@ -166,8 +209,10 @@ export default function History(): JSX.Element {
           >
             {SelectedButton == 'Code' ? (
               <CodeView code={currentCode} codeLang={codeLanguage} />
+            ) : currentMap.mapImg != '' ? (
+              <img className={styles.mapImg} src={currentMap.mapImg} />
             ) : (
-              <img className={styles.mapImg} src={mapImage} />
+              <div className={styles.mapLoad}>Loading....</div>
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.tsx]: Change `Loading...` -> `Map image not found`

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 2]
```diff
@@ -74,7 +84,33 @@ export default function History(): JSX.Element {
     });
     completeMapHistory.forEach(mapData => {
       if (mapData.id == id) {
-        setCurrentMap(JSON.parse(mapData.map));
+        setCurrentCommitMessage(mapData.message);
+        if (mapImagesByCommitIds.some(obj => obj.CommitId == id)) {
+          setCurrentMap({
+            map: JSON.parse(mapData.map),
+            mapImg: mapImagesByCommitIds.find(obj => obj.CommitId === id).Image,
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.tsx]:
```suggestion
mapImagesByCommitIds.find(obj => obj.CommitId === id)
              ?.Image as string
```

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 3]
```diff
@@ -103,8 +139,14 @@ export default function History(): JSX.Element {
       }
       toast.success(` Loaded commit - ${currentCommitMessage}`);
       navigate('/dashboard', { replace: true });
-    } else if (SelectedButton == 'Map' && currentMap.length != 0) {
+    } else if (
+      SelectedButton == 'Map' &&
+      currentMap.mapImg != '' &&
+      currentMap.map !== []
```
Comment (Ram-20062003) [inline src/components/CommitHistory/HistoryMain/History.tsx]:
```suggestion
      currentMap.map.length !== 0
```

File: src/components/BattleTV/BattleTV.tsx [hunk 4]
```diff
@@ -127,10 +135,12 @@ function PaginatedItems() {
                       }
                     </div>
                     <div className={[styles.username, styles.right].join(' ')}>
-                      {match.user2.username}
+                      {match.user2 !== null
+                        ? match.user2.username
```
Comment (Ram-20062003) [inline src/components/BattleTV/BattleTV.tsx]:
```suggestion
                        ? match.user2?.username
```

File: src/components/MapDesigner/MapDesigner.tsx [hunk 5]
```diff
@@ -103,12 +166,17 @@ const MapDesigner: React.FunctionComponent = () => {
       <div className={styles.mapDesignerContainer}>
         <MapDesignerComponent
           saveMapCallback={saveMapCallback}
+          {...MapDesignerUtils.setLocalStorageKey(
+            props.pageType == 'MapDesigner'
+              ? 'cc-map-designer-map'
+              : 'dc-map-designer-map',
+          )}
```
Comment (Ram-20062003) [inline src/components/MapDesigner/MapDesigner.tsx]: Put these in useEffect after success api call (getLatestMap)in it.

File: src/pages/Dashboard/Dashboard.tsx [hunk 6]
```diff
@@ -111,6 +134,16 @@ export default function Dashboard(): JSX.Element {
         dispatch(loggedIn());
       }
     });
+    if (dailyChallenge.challType == '') {
```
Comment (Ram-20062003) [inline src/pages/Dashboard/Dashboard.tsx]: Remove this condition check, it is redundant

File: src/store/DailyChallenge/dailyChallenge.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  dailyChallenge: {
+    challName: '',
+    description: '',
+    chall: {
+      cpp: ' ',
+      java: ' ',
+      python: ' ',
+      image: ' ',
+    },
+    challType: '',
```
Comment (Ram-20062003) [inline src/store/DailyChallenge/dailyChallenge.ts]:
```suggestion
    challType: '' as ChallengeType,
```

File: src/main.module.css [hunk 8]
```diff
@@ -23,6 +23,9 @@
   background: rgba(30, 30, 30, 0.87);
 }
 
+
 body {
   background-color: rgba(30, 30, 30, 0.87);
+  overflow-y: hidden;
 }
+
```
Comment (vigneshd332) [inline src/main.module.css]: double EOF? 🤨 🤨

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Review by vigneshd332 — APPROVED** [review]

**Pull request merged**